src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -1646,18 +1646,24 @@ private IForLoopOperation CreateBoundForStatementOperation(BoundForStatement bou
                 HashSet<DiagnosticInfo>? useSiteDiagnostics = null;
                 var compilation = (CSharpCompilation)_semanticModel.Compilation;
 
+                var iDisposable = enumeratorInfoOpt.IsAsync
+                                    ? compilation.GetWellKnownType(WellKnownType.System_IAsyncDisposable)
+                                    : compilation.GetSpecialType(SpecialType.System_IDisposable);
+
+                var enumeratorImplementsDisposable = enumeratorInfoOpt.NeedsDisposal
+                                                     && compilation.Conversions.ClassifyImplicitConversionFromType(enumeratorInfoOpt.GetEnumeratorMethod.ReturnType,
```

Comment:
>enumeratorInfoOpt.GetEnumeratorMethod [](start = 115, length = 37)

It looks like we dropped a null check for this value.

Reply:
It was never supposed to be able to contain a null value. The enumeratorInfo has nullable enabled.

Reply:
> It was never supposed to be able to contain a null value. The enumeratorInfo has nullable enabled.

It looks to me that definition of ForEachEnumeratorInfo is in nullable disabled region.

---
In reply to: [533648164](https://github.com/dotnet/roslyn/pull/49481#discussion_r533648164) [](ancestors = 533648164)